packages/sdk/src/browser-wallets/leather/index.ts

```diff
@@ -0,0 +1,148 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  BrowserWalletNetworkMismatchError,
+  BrowserWalletNotInstalledError,
+  getAddressFormat,
+  getNetworkByAddress,
+  OrditSDKError,
+} from "../..";
+import { BrowserWalletNetwork } from "../../config/types";
+import type { BrowserWalletSignResponse, WalletAddress } from "../types";
+import {
+  LeatherGetAddresses,
+  LeatherSignMessage,
+  LeatherSignMessageOptions,
+  LeatherSignPsbt,
+  LeatherSignPSBTOptions,
+} from "./types";
+import { leatherRequest } from "./utils";
+
+function isInstalled() {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+
+  return typeof window.LeatherProvider !== "undefined";
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Leather not installed");
+  }
+
+  const res = await leatherRequest<LeatherGetAddresses>("getAddresses", {
+    network,
+  });
+
+  const addresses = res.addresses.filter(
+    (address) => address.type === "p2tr" || address.type === "p2wpkh",
+  );
+
+  // Hacky validation: there's no parameter to specify the network value when getting the address.
+  // TODO: Remove this if the wallet already supports that parameter.
+  if (getNetworkByAddress(addresses[0].address) !== network) {
+    throw new BrowserWalletNetworkMismatchError(
+      "Leather network mismatch, please switch it manually",
+    );
+  }
+
+  return addresses.map((v) => ({
+    publicKey: v.publicKey!,
+    address: v.address,
+    format: getAddressFormat(v.address, network),
```

Comment:
there is already `address.type` used above in filtering, is it different from `getAddressFormat`? or should we validate that both should be the same as well?

Reply:
yeah that's different, the address.type above is coming from leather api, need to filter only for p2tr(ordinal) & p2wpkh(payment) -> https://leather.gitbook.io/developers/bitcoin/connect-users/get-addresses, `getAddressFormat` is used to validate the address and for getting the sdk standard type (ex: p2tr -> taproot)

Reply:
yeah we can, is that oke @Nanosync ?

Reply:
yes, we can use this variable, won't need to revalidate the address since we're already filtering `address.type`